setup.py

```diff
@@ -89,7 +89,7 @@ def read_response(cmd):
                 try:
                     gdalinfo_path = None
                     for path in os.getenv("PATH", "").split(os.pathsep):
-                        matches = list(Path(path).glob("**/gdalinfo*"))
+                        matches = list(Path(path).glob("**/gdalinfo.exe"))
```

Comment:
I made this change because in my latest conda env, this was not working (while gdalinfo is available in a conda env on windows). The reason for this is that the current glob also found other gdalinfo occurences:

```
> c:\users\joris\scipy\pyogrio\setup.py(93)<module>()
-> matches = list(Path(path).glob("**/gdalinfo*"))
(Pdb) path
'C:\\Users\\joris\\miniconda3\\envs\\geo-dev'
(Pdb) n
> c:\users\joris\scipy\pyogrio\setup.py(94)<module>()
-> if matches:
(Pdb) matches
[WindowsPath('C:/Users/joris/miniconda3/envs/geo-dev/Lib/site-packages/osgeo_utils/samples/gdalinfo.py'), WindowsPath('C:/Users/joris/miniconda3/envs/geo-dev/Lib/site-packages/osgeo_utils/samples/__pycache__/gdalinfo.cpython-310.pyc'), WindowsPath('C:/Users/joris/miniconda3/envs/geo-dev/Library/bin/gdalinfo.exe')]
```